src/ribasim_nl/ribasim_nl/model.py

```diff
@@ -0,0 +1,143 @@
+# %%
+import geopandas as gpd
+import pandas as pd
+import ribasim
+from ribasim import Model, Network
+from shapely.geometry import LineString, Point
+
+from ribasim_nl.case_conversions import pascal_to_snake_case
+
+# TODO: get this from ribasim somehow
+CLASS_TABLES = {
+    "Basin": ["static", "profile", "state"],
+    "LinearResistance": ["static"],
+    "ManningResistance": ["static"],
+    "Pump": ["static"],
+    "Outlet": ["static"],
+    "Terminal": ["static"],
+    "FlowBoundary": ["static"],
+    "LevelBoundary": ["static", "time"],
+    "FractionalFlow": ["static"],
+    "TabulatedRatingCurve": ["static"],
+}
+
+TABLES = [
+    j
+    for r in [
+        [f"{pascal_to_snake_case(k)}.{i}.df" for i in v]
+        for k, v in CLASS_TABLES.items()
+    ]
+    for j in r
+]
+
+
+def get_table(model: Model, table: str = "basin.static.df"):
```

Comment:
Would be nice if we don't need this.
In Ribasim Python `model.nodes()` gives all node types, and per node type there is `node.tables()` to iterate over the tables.
Referring to tables as a string like `"basin.static.df"` looks wrong, that is Python code. Also does it make sense to have a default table argument?

Reply:
I agree, hence why my first comment and an attempt to improve this in a new issue: https://github.com/Deltares/Ribasim-NL/issues/69